prometheus/push/push.go

```diff
@@ -170,7 +170,7 @@ func (p *Pusher) Grouping(name, value string) *Pusher {
 
 // Client sets a custom HTTP client for the Pusher. For convenience, this method
 // returns a pointer to the Pusher itself.
-func (p *Pusher) Client(c *http.Client) *Pusher {
+func (p *Pusher) Client(c HTTPClient) *Pusher {
```

Comment:
The doc comment needs to be amended. Suggestion for additional text:

Pusher only needs one method of the custom HTTP client: Do(*http.Request). Thus, rather than requiring a fully fledged http.Client, the provided client only needs to implement the HTTPDoer interface. Since *http.Client naturally implements that interface, it can still be used normally.